test/e2e/create_cluster_test.go

```diff
@@ -56,6 +56,47 @@ func TestAWSCreateCluster(t *testing.T) {
 	e2eutil.EnsureNoCrashingPods(t, ctx, client, hostedCluster)
 }
 
+// TestKubeVirtCreateCluster implements a test that mimics the operation described in the
+// HyperShift quick start (creating a basic guest cluster).
+//
+// This test is meant to provide a first, fast signal to detect regression; it
+// is recommended to use it as a PR blocker test.
+func TestKubeVirtCreateCluster(t *testing.T) {
+	t.Parallel()
+	g := NewWithT(t)
+
+	ctx, cancel := context.WithCancel(testContext)
+	defer cancel()
+
+	client := e2eutil.GetClientOrDie()
+
+	clusterOpts := globalOpts.DefaultClusterOptions()
+
+	hostedCluster := e2eutil.CreateKubeVirtCluster(t, ctx, client, clusterOpts, globalOpts.ArtifactDir)
+
+	// Get the newly created nodepool
+	nodepool := &hyperv1.NodePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: hostedCluster.Namespace,
+			Name:      hostedCluster.Name,
+		},
+	}
+	err := client.Get(testContext, crclient.ObjectKeyFromObject(nodepool), nodepool)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get nodepool")
+	t.Logf("Created nodepool. Namespace: %s, name: %s", nodepool.Namespace, nodepool.Name)
+
+	// Wait for hosted cluster to become ready
+	e2eutil.WaitForHostedClusterAvailable(t, testContext, client, hostedCluster)
+
+	// Wait for kubevirt machines to come online
+	e2eutil.WaitForKubeVirtMachines(t, testContext, client, hostedCluster, *nodepool.Spec.NodeCount)
```

Comment:
why is this not using `WaitForNReadyNodes`? if there's a reason please add a TODO

Reply:
Yep, i'll write a comment for why this is necessary.

The issue at the moment is that we don't have guest cluster access because there's no proper ingress. We're working on sorting out options for ingress at the moment. Right now the best we can do to get some preliminary test results of the KubeVirt platform is to introspect the HostedCluster's related CRs on the management cluster. It's not great, but gives us some confidence that at least the KubeVirt VMs launch and that the HostedCluster is marked as being available.